Clarify running nodes (#49)
etherlinkcom/docs: docs/network/evm-nodes.md

---
title: Running an Etherlink EVM node
---

> The Etherlink EVM nodes are responsible for maintaining a copy of the Etherlink context and creating and applying new blocks that process EVM transactions.

Reviewer: Here we explain the worflow for the observer EVM node. The observer don't create blocks,
```suggestion
The Etherlink EVM nodes are responsible for maintaining a copy of the Etherlink context and applying new blueprints that process EVM transactions.
```

## Prerequisites

- Make sure you understand the interaction between different nodes as described in [Etherlink architecture](./architecture).
- Run an Etherlink Smart Rollup node as described in [Running an Etherlink Smart Rollup node](./smart-rollup-nodes).

> - Get the Etherlink installer kernel (`installer.hex` file), which you can build yourself as described in [Building the Etherlink kernel](./smart-rollup-nodes#building-the-etherlink-kernel) or download here: [installer.hex](/files/installer.hex).

Reviewer: This is a prerequisites if you want to start from genesis and want contains the complete history.

Reviewer: Which is what you describe after so my comment is not really important haha

## Running an Etherlink EVM node

Follow these steps to run the EVM node:

1. Get a built version of the EVM node binary, named `octez-evm-node`.
Octez does not yet provide a binary build of the EVM node as part of its binary distribution or in the `tezos/tezos` docker image, so you must build it yourself from the latest commit from the Octez source code.
See [Installing Octez](https://tezos.gitlab.io/introduction/howtoget.html).
1. Set the `sr_node_observer_rpc` environment variable to the URL to the Smart Rollup node you set up in the previous section, such as `http://localhost:8932`.

> 1. Set the `evm_observer_dir` environment variable to the directory where the node should store its local data.

Reviewer: ```suggestion
1. Set the `evm_observer_dir` environment variable to the directory where the node should store its local data. Default is `$HOME/.octez-evm-node`.
```

1. Initialize the node by running this command:

   ```bash
   octez-evm-node init config --devmode \
     --data-dir $evm_observer_dir --rollup-node-endpoint $sr_node_observer_rpc \
     --preimages-endpoint https://snapshots.eu.tzinit.org/etherlink-mainnet/wasm_2_0_0 \
     --evm-node-endpoint https://node.mainnet.etherlink.com
   ```

>    This command uses preimages hosted online to set up the kernel based on the installer kernel that you built.

Reviewer: This is also useful for future upvoted kernel. I think this is enough to make the point.

```suggestion
   This command uses preimages hosted online.
```

   It's safe to use these preimages, but if you don't trust them, you can move the contents of the `wasm_2_0_0/` directory that was created when you built the installer kernel to the local data directory.

>    However, in this case, the kernel is not updated automatically, so you must manually update this directory with the preimages of every kernel voted by the community and deployed on Etherlink after that.

Reviewer: The kernel will try to upload and simply block until the preimages are copied.
```suggestion
   However, in this case, you must manually update this directory with the preimages of every kernel voted by the community and deployed on Etherlink after that.
```

---
title: Running an Etherlink EVM node
---

The Etherlink EVM nodes are responsible for maintaining a copy of the Etherlink context and applying new blueprints that process EVM transactions.

## Prerequisites

- Make sure you understand the interaction between different nodes as described in [Etherlink architecture](./architecture).
- Run an Etherlink Smart Rollup node as described in [Running an Etherlink Smart Rollup node](./smart-rollup-nodes).
Public Smart Rollup nodes for Etherlink are not yet available, so you must run your own if you want to participate in the Etherlink network.
- Get the Etherlink installer kernel (`installer.hex` file), which you can build yourself as described in [Building the Etherlink kernel](./building-kernel) or download here: [installer.hex](/files/installer.hex).

> ## Running an Etherlink EVM node

Reviewer: Now that we have smart rollup snapshot we can start an Etherlink EVM node without the initial kernel.
But maybe in another MR ?

Reviewer: with the `init from rollup node` command

Author: That's new to me. I'll make a note to handle that in a future PR.

Reviewer: Do you uses issue for doc PR ?
if yes, could you turn this into a follow-up ?

Author: Ok, here's the issue for starting the EVM node without the initial kernel: https://github.com/etherlinkcom/docs/issues/63

Follow these steps to run the EVM node:

1. Get a built version of the EVM node binary, named `octez-evm-node`.
Octez does not yet provide a binary build of the EVM node as part of its binary distribution or in the `tezos/tezos` docker image, so you must build it yourself from the latest commit from the Octez source code.
See [Installing Octez](https://tezos.gitlab.io/introduction/howtoget.html).
1. Set the `sr_node_observer_rpc` environment variable to the URL to the Smart Rollup node you set up in the previous section, such as `http://localhost:8932`.
1. Set the `evm_observer_dir` environment variable to the directory where the node should store its local data.
The default is `$HOME/.octez-evm-node`.
1. Initialize the node by running this command:

   ```bash
   octez-evm-node init config --devmode \
     --data-dir $evm_observer_dir --rollup-node-endpoint $sr_node_observer_rpc \
     --preimages-endpoint https://snapshots.eu.tzinit.org/etherlink-mainnet/wasm_2_0_0 \
     --evm-node-endpoint https://node.mainnet.etherlink.com
   ```

   This command uses preimages hosted online.

>    It's safe to use these preimages, but if you don't trust them, you can move the contents of the `wasm_2_0_0/` directory that was created when you built the installer kernel to the local data directory.

Reviewer 2: Missing a link to the new page about building the kernel.

Besides, I’ll reword with

```suggestion
   It's safe to use these preimages, as their integrity is verified by the node. However, if you don't want to rely on a third-party service, you can move the contents of the `wasm_2_0_0/` directory that was created when you built the installer kernel to the local data directory.
```

Author: Changed to:

> This command uses preimages hosted online. It's safe to use these preimages because the node verifies them. If you don't want to use third-party preimages, you can build the kernel yourself and move the contents of the wasm_2_0_0/ directory to the local data directory; see [Building the Etherlink kernel](http://localhost:3001/network/building-kernel). However, in this case, you must manually update this directory with the preimages of every kernel voted by the community and deployed on Etherlink after that.
